DataFormats/L1TParticleFlow/interface/bit_encoding.h

```diff
@@ -58,4 +58,34 @@ inline void l1pf_pattern_unpack(const ap_uint<NB> data[], T objs[N]) {
   }
 }
 
+template <unsigned int N, unsigned int OFFS = 0, typename T, int NB>
+inline void l1pf_pattern_pack_slim(const T objs[N], ap_uint<NB> data[]) {
+#ifdef __SYNTHESIS__
+#pragma HLS inline
+#pragma HLS inline region recursive
+#endif
+  assert(T::BITWIDTH_SLIM <= NB);
+  for (unsigned int i = 0; i < N; ++i) {
+#ifdef __SYNTHESIS__
+#pragma HLS unroll
+#endif
+    data[i + OFFS] = objs[i].pack_slim();
+  }
+}
+
+template <unsigned int N, unsigned int OFFS = 0, typename T, int NB>
+inline void l1pf_pattern_unpack_slim(const ap_uint<NB> data[], T objs[N]) {
+#ifdef __SYNTHESIS__
+#pragma HLS inline
+#pragma HLS inline region recursive
+#endif
+  assert(T::BITWIDTH_SLIM <= NB);
+  for (unsigned int i = 0; i < N; ++i) {
+#ifdef __SYNTHESIS__
+#pragma HLS unroll
```

Comment:
Out of curiosity, are these pragmas understood at the CMSSW level? The checking script may not have caught it, but it may generate unknown pragma warnings in central CMSSW

Reply:
they're protected by the `#ifdef __SYNTHESIS__`, which is defined by HLS during synthesis but not by gcc ir clang during CMSSW compilation.